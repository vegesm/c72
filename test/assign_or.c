main() {
    extern iarr;
    int iarr[];
    int a,b;
    char c, d, carr[4];

    a = 5;
    printf("a | 3 = %d\n", a =| 3);
    printf("a = %d\n", a);

    a = 5; b =3;
    printf("a | b = %d\n", a =| b);
    printf("a = %d\n", a);

    printf("iarr[0] | b = %d\n", *iarr =| b);
    printf("iarr[0] = %d\n", iarr[0]);

    printf("iarr[1] | b = %d\n", iarr[1] =| b);
    printf("iarr[1] = %d\n", iarr[1]);

    c = 5;
    printf("c | 3 = %d\n", c =| 3);
    printf("c = %d\n", c);

    c = 5; d =3;
    printf("c | d = %d\n", c =| b);
    printf("c = %d\n", c);

    carr[0] = 5;
    carr[1] = -10;

    printf("carr[0] | d = %d\n", *carr =| d);
    printf("carr[0] = %d\n", carr[0]);

    printf("carr[0] | 128 = %d\n", *carr =| 128);
    printf("carr[0] = %d\n", carr[0]);
    printf("carr[1] = %d\n", carr[1]);

    printf("carr[1] | d = %d\n", carr[1] =| d);
    printf("carr[1] = %d\n", carr[1]);
}

iarr[] 0, 5, 10, 128;

