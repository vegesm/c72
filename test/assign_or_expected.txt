a | 3 = 7
a = 7
a | b = 7
a = 7
iarr[0] | b = 3
iarr[0] = 3
iarr[1] | b = 7
iarr[1] = 7
c | 3 = 7
c = 7
c | d = 7
c = 7
carr[0] | d = 7
carr[0] = 7
carr[0] | 128 = -121
carr[0] = -121
carr[1] = -10
carr[1] | d = -9
carr[1] = -9
